### DISCUSSION TITLE
Do we have code completion for eslint.config.js?

### DISCUSSION THREAD

**Discussion by GCSBOSS** [eslint/eslint#17551] [Q&A] [answered]
Is there any JSDoc snippet/type I can slap on my `eslint.config.js` to have code completion for existing eslint rules/options?

**Comment by nzakas** [comment, marked as answer]
No. Even if we were to provide a type for flat config, it would not include rules and options because those are loaded dynamically at runtime.

**Reply by GCSBOSS** [reply]
Doesn't a d.ts file cover that gap? Or is it too much of a burden to maintain that?

**Reply by nzakas** [reply]
It would be complex to generate that for all core rules as we'd need to extract that information from the rule schema (which is JSON Schema). And then you still wouldn't have any coverage for the thousands of custom rules people have made and published in plugins.

**Reply by GCSBOSS** [reply]
Got it. 
I don't really use plugins but in that case, am I correct to think that should be responsibility for the plugin authors to have their own types?

Having said that. Do you think it's worth it to open an issue/start a PR?
